src/spyglass/lfp/analysis/v1/lfp_band.py

```diff
@@ -109,57 +137,133 @@ def set_lfp_band_electrodes(
+                + "lfp data to avoid aliasing."
+            )
+        if lfp_band_sampling_rate > lfp_sampling_rate:
+            raise ValueError(
+                "lfp_band_sampling_rate must be less than or equal to the "
+                + "lfp sampling rate"
+            )
+        if lfp_band_sampling_rate <= 0:
+            raise ValueError("lfp_band_sampling_rate must be greater than 0")
+        decimation = lfp_sampling_rate // lfp_band_sampling_rate
 
-        key = dict(
+        master_key = dict(
             nwb_file_name=nwb_file_name,
             lfp_merge_id=lfp_merge_id,
             filter_name=filter_name,
             filter_sampling_rate=lfp_sampling_rate,
             target_interval_list_name=interval_list_name,
             lfp_band_sampling_rate=lfp_sampling_rate // decimation,
         )
-        # insert an entry into the main LFPBandSelectionTable
-        self.insert1(key, skip_duplicates=True)
-
-        key["lfp_electrode_group_name"] = lfp_part_table.fetch1(
+        # nwb_file_name, lfp_electrode_group_name, electrode_group_name, electrode_id, reference_elect_id
+        part_keys = []
+        lfp_electrode_group_name = lfp_part_table.fetch1(
             "lfp_electrode_group_name"
         )
-        # iterate through all of the new elements and add them
-        for e, r in zip(electrode_list, ref_list):
-            elect_key = (
-                LFPElectrodeGroup.LFPElectrode
-                & {
+
+        # get the electrode group names for the electrodes
+        electrode_group_name_list, check_electrode_id_list = (
+            LFPElectrodeGroup.LFPElectrode()
+            & [
+                {
                     "nwb_file_name": nwb_file_name,
-                    "lfp_electrode_group_name": key["lfp_electrode_group_name"],
-                    "electrode_id": e,
+                    "lfp_electrode_group_name": lfp_electrode_group_name,
+                    "electrode_id": electrode_id,
                 }
-            ).fetch1("KEY")
-            for item in elect_key:
-                key[item] = elect_key[item]
-            query = Electrode & {
-                "nwb_file_name": nwb_file_name,
-                "electrode_id": e,
-            }
-            key["reference_elect_id"] = r
-            self.LFPBandElectrode().insert1(key, skip_duplicates=True)
+                for electrode_id in electrode_list
+            ]
+        ).fetch("electrode_group_name", "electrode_id", order_by="electrode_id")
+
+        # check that the electrode ids match in order to avoid
+        # inserting the wrong electrode group names
+        if not np.array_equal(electrode_list, check_electrode_id_list):
+            raise ValueError(
+                "Electrode IDs do not match the electrode IDs in the "
+                + "LFPElectrodeGroup table"
+            )
+
+        for electrode_id, reference_elect_id, electrode_group_name in zip(
```

Comment:
List comp here would save you the declaration of part keys above. Even if preserved as a for loop, I would move the `part_keys = []` down to be paired with the block that appends to it